crates/db_views/src/site/site_view.rs

```diff
@@ -2,25 +2,37 @@ use crate::structs::SiteView;
 use diesel::{ExpressionMethods, JoinOnDsl, OptionalExtension, QueryDsl, SelectableHelper};
 use diesel_async::RunQueryDsl;
 use lemmy_db_schema::{
-  schema::{local_site, local_site_rate_limit, site},
+  schema::{instance, local_site, local_site_rate_limit, site},
   utils::{get_conn, DbPool},
 };
-use lemmy_utils::error::{LemmyErrorType, LemmyResult};
+use lemmy_utils::{
+  build_cache,
+  error::{LemmyError, LemmyErrorType, LemmyResult},
+  CacheLock,
+};
+use std::sync::{Arc, LazyLock};
 
 impl SiteView {
   pub async fn read_local(pool: &mut DbPool<'_>) -> LemmyResult<Self> {
-    let conn = &mut get_conn(pool).await?;
-    Ok(
-      site::table
-        .inner_join(local_site::table)
-        .inner_join(
-          local_site_rate_limit::table.on(local_site::id.eq(local_site_rate_limit::local_site_id)),
-        )
-        .select(Self::as_select())
-        .first(conn)
-        .await
-        .optional()?
-        .ok_or(LemmyErrorType::LocalSiteNotSetup)?,
-    )
+    static CACHE: CacheLock<SiteView> = LazyLock::new(build_cache);
```

Comment:
Remove caching from `LocalSite::read`, and replace every case of it with this one. Possibly even remove that function or make it non-public.

Reply:
See my comment here: https://github.com/LemmyNet/lemmy/pull/5515#discussion_r2011763469